src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,92 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import duke.tasklist.TaskList;
+
+import duke.exception.DukeException;
+
+public class Parser {
+
+    public static Boolean FeedDuke(String input, TaskList taskList) throws DukeException {
+        if (input.contains("bye")) {
+            return true;
+            
+        } else if (input.contains("list")) {
+            taskList.printList();
+            return false;
+
+        } else if (input.contains("unmark")) {
+            int index = Integer.parseInt(input.replace("unmark", "").trim()) - 1;
+            taskList.unMarkTask(index);
+            return false;
+
+        } else if (input.contains("mark")) {
+            int index = Integer.parseInt(input.replace("mark", "").trim()) - 1;
+            taskList.markTask(index);
+            return false;
+
+        } else if (input.contains("todo")) {
+            String content = input.replace("todo", "").trim();
+            if (content.length() == 0) {
+                throw new DukeException("\t☹ OOPS!!! The description of a todo cannot be empty.");
```

Comment:
Instead of multiple `DukeException` objects with similar error messages, consider making a child class of `DukeException` to handle similar exception issues.